src/algo/depth-to-rgb-calibration/valid-scene.cpp

```diff
@@ -679,5 +679,151 @@ bool optimizer::is_scene_valid()
+
+    std::vector<double> val_per_dir[DIRECTIONS];
+
+    for (auto i = 0; i < subpixels_x.size(); i++)
+    {
+        auto dir = (int)directions[i] - 1;
+        auto val = subpixels_x[i] * dir_vecs[dir].x + subpixels_y[i] * dir_vecs[dir].y;
+        val_per_dir[dir].push_back(val);
+    }
+
+    double std_per_dir[DIRECTIONS] = { 0 };
+    bool std_bigger_than_th[DIRECTIONS] = { false };
+
+    for (auto i = 0; i < DIRECTIONS; i++)
+    {
+        auto curr_dir = val_per_dir[i];
+        double sum = std::accumulate(curr_dir.begin(), curr_dir.end(), 0.0);
+        double mean = sum / curr_dir.size();
+
+        double dists_sum = 0;
+        std::for_each(curr_dir.begin(), curr_dir.end(), [&](double val) {dists_sum += (val - mean)*(val - mean); });
+
+        // The denominator in the 'Sample standard deviation' formula is N − 1 vs 'Population standard deviation' that is N
+        // https://en.wikipedia.org/wiki/Standard_deviation
+        // we use 'Sample standard deviation' as Matlab
+        auto stdev = sqrt(dists_sum / (curr_dir.size() - 1));
+        std_per_dir[i] = stdev / diag_length;
+        std_bigger_than_th[i] = std_per_dir[i] > p.dir_std_th[i];
+    }
+
+    bool valid_directions[DIRECTIONS] = { false };
+
+    for (auto i = 0; i < DIRECTIONS; i++)
+    {
+        valid_directions[i] = dirs_with_enough_edges[i] && std_bigger_than_th[i];
+    }
+    auto valid_directions_sum = std::accumulate(&valid_directions[0], &valid_directions[DIRECTIONS], 0);
+
+    auto edges_dir_spread = valid_directions_sum > p.minimal_full_directions;
+
+    if (!edges_dir_spread)
+    {
+        AC_LOG(DEBUG, "Scene is not valid since there is not enough edge direction spread");
+        return edges_dir_spread;
+    }
+
+    if (p.require_orthogonal_valid_dirs)
+    {
+        auto valid_even = true;
+        for (auto i = 0; i < DIRECTIONS; i += 2)
+        {
+            valid_even &= valid_directions[i];
+        }
+
+        auto valid_odd = true;
+        for (auto i = 1; i < DIRECTIONS; i += 2)
+        {
+            valid_odd &= valid_directions[i];
+        }
+        auto orthogonal_valid_dirs = valid_even || valid_odd;
+
+        if (!orthogonal_valid_dirs)
+            AC_LOG(DEBUG, "Scene is not valid since there is no at least two orthogonal directions that have enough spread edges");
+
+        return edges_dir_spread && orthogonal_valid_dirs;
+    }
+
+    return edges_dir_spread;
 }
+
+bool check_saturation(const std::vector< ir_t >& ir_frame,
+    size_t width,
+    size_t height,
+    const params& p)
+{
+    int saturated_pixels = 0;
+
+    for (auto&& i : ir_frame)
+    {
+        if (i >= p.saturation_value)
+            saturated_pixels++;
+    }
+
+    auto saturated_pixels_ratio = (double)saturated_pixels / (double)(width*height);
+
+    if (saturated_pixels_ratio < p.saturation_ratio_th)
+        AC_LOG(DEBUG, "Scene is not valid since the saturated is above threshold");
```

Comment:
Maybe make this an error, so it's visible in the Viewer log? Same for the other errors that will make a scene not pass...